turtlesim/tutorials/draw_square.cpp

```diff
@@ -1,163 +1,230 @@
-#include <boost/bind.hpp>
-#include <ros/ros.h>
-#include <turtlesim/Pose.h>
-#include <geometry_msgs/Twist.h>
-#include <std_srvs/Empty.h>
+#include <rclcpp/rclcpp.hpp>
+#include <turtlesim/msg/pose.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <std_srvs/srv/empty.hpp>
 
-turtlesim::PoseConstPtr g_pose;
-turtlesim::Pose g_goal;
+#include <math.h>
+
+using namespace std::chrono_literals;
+
+#define PI 3.141592
 
 enum State
 {
-  FORWARD,
+  FORWARD = 0,
   STOP_FORWARD,
   TURN,
   STOP_TURN,
 };
 
-State g_state = FORWARD;
-State g_last_state = FORWARD;
-bool g_first_goal_set = false;
-
-#define PI 3.141592
-
-void poseCallback(const turtlesim::PoseConstPtr& pose)
+class DrawSquare : public rclcpp::Node
```

Comment:
As of here the patch contains still largely unnecessary changes. Please keep the patch as close as possible to the upstream branch.

This applies to pretty much everything below.

Reply:
Accepted